### PR TITLE
Promisification and major abstractions

### PR DESCRIPTION
**Pull request by adamlawrencium**
Lots of changes that probably should've been in separate pull requests... : 
- replaced callbacks and async.js functions with ES6 promises
- added abstractStrategy class
- added DataHandler class to abstract away pulling exchange data
- moved old code into 'old' folder
- added some console fun

### CODE REVIEW

**Comment by BirchellE** [comment]
Oooh boy those classes look cool

**Comment by BirchellE** [comment]
I think the old code deserves Medicaid instead of being shoved away in the "old folder". Who are you Mitt Romney?

File: lib/ascii-logo.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Welcome to...
+   ____       __  _
+  / __ \_____/ /_(_)___ _
+ / / / / ___/ __/ / __ `/
+/ /_/ (__  ) /_/ / /_/ /
+\____/____/\__/_/\__,_/...
```
Comment (BirchellE) [inline lib/ascii-logo.txt:6]: 10/10

**Comment by adamlawrencium** [comment]
Hmm, good point. Will add 'administer Medicaid' to the todos.

**Pull request merged**